common/src/main/java/com/wynntils/features/trademarket/TradeMarketQuickSearchFeature.java

```diff
@@ -0,0 +1,183 @@
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraft.world.inventory.Slot;
+import net.neoforged.bus.api.EventPriority;
+import net.neoforged.bus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.TRADEMARKET)
+public class TradeMarketQuickSearchFeature extends Feature {
+    @Persisted
+    public final Config<Boolean> instantSearch = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> autoCancel = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> hidePrompt = new Config<>(false);
+
+    @RegisterKeyBind
+    private final KeyBind quickSearchKeyBind = new KeyBind(
+            "Quick Search TM",
+            GLFW.GLFW_MOUSE_BUTTON_MIDDLE,
+            InputConstants.Type.MOUSE,
+            true,
+            null,
+            this::tryQuickSearch);
+
+    // Test in TradeMarketQuickSearchFeature_TYPE_TO_CHAT_PATTERN
+    private static final Pattern TYPE_TO_CHAT_PATTERN =
+            Pattern.compile("^§5(\uE00A\uE002|\uE001) Type the .* or type 'cancel' to cancel:");
+
+    // The TM is very peculiar...
+    // 'Emerald Pouch' results in all Tiers of Pouches, to find a specific tier one needs to search only the tier.
+    // e.g. "Emerald Pouch [Tier V]" results in nothing, while "[Tier V]" results in the desired Pouches.
+    // (This is also the reason replaceFirst is used,
+    // as "Emerald Pouch" would be the first match, and "[Tier V]" would not get removed.)
+
+    // '\\[.*?\\]' Many items have square brackets at the end displaying some kind of information.
+    // e.g. durability on crafted items [50%], uses on consumables [3/3], stars on materials and ingredients [✫✫✫]
+
+    // '\\[\\uE000-\\uF8FF\\]' This range contains special chars used by Wynncraft.
+    // e.g. The prof icon at the beginning of Gathering tools
+
+    // '\\u2B21' Just the hexagon char "⬡" used for Shinies.
+    // 'Unidentified \\u2B21' As of now there is no known way of matching unided shinies, shiny is the next best thing.
+    private static final Pattern CUT_PATTERN =
+            Pattern.compile("(Emerald Pouch|\\[.*?\\]|[\\uE000-\\uF8FF]+|Unidentified \\u2B21|\\u2B21)\\s*");
+
+    // Similarly to the Pouches, the elemental/skill potions cannot be found by their full name (Other potions can).
+    // If we are searching such a potion, we simply search the second part. '.' matches the icon and '\w+' the skill.
+    // e.g. "Potion of ✤ Strength [2/2]" -> "Potion of ✤ Strength" -> "✤ Strength"
+    private static final Pattern POTION_PATTERN = Pattern.compile("^Potion of (. \\w+)");
+
+    // 'À' this char is behind all the items names and needs to be trimmed before we search.
+    // We cannot use WynnUtils.normalizeBadString() as this messes up dungeon keys.
+    // e.g. "CorruptedÀÀÀGalleon'sÀÀÀGraveyard KeyÀ" -> "CorruptedÀÀÀGalleon'sÀÀÀGraveyard Key"
```

Comment:
What does this mean? That you need to search for "CorruptedÀÀÀGalleon'sÀÀÀGraveyard" instead of "Corrupted Galleon's Graveyard"? That surprises me, but if it really is the case then I guess that is what it is.

Or does it mean that normalizeBadString does not work properly? I would have assumed that it would generate "Corrupted Galleon's Graveyard" from "CorruptedÀÀÀGalleon'sÀÀÀGraveyard"  -- is that not so?

Reply:
Yeah the market is weird like that, you need to search with the ÀÀÀ but not the À at the end, took me ages to figure out because I was using normalizeBadStrings at first.